homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,389 @@
+        """Return the name of the media_player."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the media_player."""
+        return self._state
+
+    @property
+    def is_volume_muted(self):
+        """Return the state of the muting function."""
+        return self._muted
+
+    @property
+    def source(self):
+        """Return currently selected input source."""
+        return self._audio_source
+
+    @property
+    def source_list(self):
+        """Return a list of available Presets."""
+        return self._source_list
+
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        features = SUPPORT_CAMBRIDGE_COMMON
+        if self._audio_source == "media player":
+            features = features | SUPPORT_CAMBRIDGE_FILEMEDIA
+        if self._support_volumecontrol:
+            features = features | SUPPORT_CAMBRIDGE_PREAMP
+        return features
+
+    @property
+    def media_content_type(self):
+        """Return the media content type."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def media_image_url(self):
+        """Return image URI of current playing media."""
+        return self._album_art
+
+    @property
+    def media_artist(self):
+        """Return artist of current playing media."""
+        return self._artist
+
+    @property
+    def media_album_name(self):
+        """Return album of current playing media."""
+        return self._album
+
+    @property
+    def media_track(self):
+        """Return track number of current playing media."""
+        return self._trackno
+
+    @property
+    def media_title(self):
+        """Return title of current playing media."""
+        return self._title
+
+    @property
+    def media_position(self):
+        """Return position of current playing media in seconds."""
+        if self._state is not STATE_PLAYING:
+            return None
+        return self._position
+
+    @property
+    def media_duration(self):
+        """Return the duration of current playing media in seconds."""
+        if self._state is not STATE_PLAYING:
+            return None
+        return self._duration
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid.
+
+        Returns value from homeassistant.util.dt.utcnow().
+        """
+        if self._state in [STATE_PLAYING, STATE_PAUSED]:
+            return self._position_updated_at
+
+    @property
+    def volume_level(self):
+        """Current volume level (0..1)."""
+        return float(self._volume) / float(self._volume_max)
+
+    def clear_playlist(self):
+        """Clear the playlist."""
+        pass
+
+    def media_next_track(self):
+        """Skip to the next track, when in media player mode."""
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_next()
+            self.schedule_update_ha_state()
```

Comment:
You don't define a `should_poll` so Home Assistant will already call this method right after it is done calling this method. So please remove this line.

Reply:
Ok, I was using this wrong. I assumed update() would run at fixed periodic intervals while schedule_update_ha_state() would do an "out of band" update.
I removed this now.